### PR TITLE
MechKeys ACR60 Layout Updates

### PR DESCRIPTION
**Pull request by noroadsleft**
## Description

- adds 22 layouts
- removes 3 layouts
- renames `LAYOUT_mitchsplit` to `LAYOUT_60_ansi_split_space_split_rshift`
- adds layout/matrix diagram
- enables Community Layout support

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by fauxpark — APPROVED** [review]

**Comment by dunk2k** [comment]
Hi @noroadsleft,
I'd have to challenge addition of `60_abnt2` layout:
1. Using [image](https://i.imgur.com/55UK7s9.jpeg) as reference of PCB in question: From left to right, the last 2 switch holes of the shift/4th row are same spacing as last 2 holes of the number/1st row, as well as key stabiliser holes for a 2u key - which affirms the 2nd last switch hole on the shift row is for a 2u key, not a 1.75u key.
2. Using the measurement between the ABNT2 `.` key switch hole and ABNT2 `;` key switch hole to locate the ABNT2 `/` key shows there isn't a switch hole in the PCB for ABNT2 `/` key.
[ABNT2 reference](https://upload.wikimedia.org/wikipedia/commons/thumb/e/e4/KB_Portuguese_Brazil_ABNT2.svg/1200px-KB_Portuguese_Brazil_ABNT2.svg.png)

I'm struggling to locate official information from mechkeys.ca regarding this PCB, so images are best that's available 🫤

File: keyboards/mechkeys/acr60/matrix_diagram.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+               └──────────┴────┴────────┘
+┌────┬────┬────┬────────────────────────┬────┬────┬────┬────┐ ┐
+│40  │41  │43  │46                      │4A  │4B  │4D  │4E  │ ├─ Standard
+└────┴────┴────┴────────────────────────┴────┴────┴────┴────┘ │    or
+┌─────┬───┬─────┬───────────────────────┬─────┬───┬───┬─────┐ │  Infinity/True HHKB
+│40   │41 │43   │46                     │4A   │4B │4D │4E   │ │    (same matrix)
+└─────┴───┴─────┴───────────────────────┴─────┴───┴───┴─────┘ ┘
+┌─────┬───┬─────┬───────────────────────────┬─────┬───┬─────┐
+│40   │41 │43   │46                         │4B   │4D │4E   │ Tsangan/WKL/HHKB
+└─────┴───┴─────┴───────────────────────────┴─────┴───┴─────┘
```
Comment (dunk2k) [inline keyboards/mechkeys/acr60/matrix_diagram.md:46]:
```suggestion
└─────┴───┴─────┴───────────────────────────┴─────┴───┴─────┘
┌─────┬─────┬───────────────────────────┐
│40   │41   │46                         │                     LWKL
└─────┴─────┴───────────────────────────┘
```

File: keyboards/mechkeys/acr60/matrix_diagram.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+├───┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴───┤      ┌─────┐
+│10   │12 │13 │14 │15 │16 │17 │18 │19 │1A │1B │1C │1D │1E   │      │     │
+├─────┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴─────┤   ┌──┴┐2D  │ ISO Enter
+│20    │22 │23 │24 │25 │26 │27 │28 │29 │2A │2B │2C │2D      │   │1E │    │
+├───┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬───┬───┤   └───┴────┘
+│30 │31 │32 │33 │34 │35 │36 │37 │38 │39 │3A │3B │3C │3D │3E │
+└───┴───┴───┴───┴───┴───┴───┴───┴───┴───┴───┴───┴───┴───┴───┘
+┌───────┐
+│30     │ 2u LShift
+└───────┘
```
Comment (dunk2k) [inline keyboards/mechkeys/acr60/matrix_diagram.md:18]:
```suggestion
┌───────┐                                           ┌───────┐
│30     │ 2u LShift                       2u Rshift │3E     │
└───────┘                                           └───────┘
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/mechkeys/acr60/matrix_diagram.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+Alternate Shift Row Options:
```
Comment (dunk2k) [inline keyboards/mechkeys/acr60/matrix_diagram.md:19]:
```suggestion
Shift Row Options:
┌───────┐                                           ┌───────┐
│30     │ 2u LShift                       2u RShift │3E     │
└───────┘                                           └───────┘
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

**Comment by noroadsleft** [comment]
> 2. Using the measurement between the ABNT2 `.` key switch hole and ABNT2 `;` key switch hole to locate the ABNT2 `/` key shows there isn't a switch hole in the PCB for ABNT2 `/` key.
>        [ABNT2 reference](https://upload.wikimedia.org/wikipedia/commons/thumb/e/e4/KB_Portuguese_Brazil_ABNT2.svg/1200px-KB_Portuguese_Brazil_ABNT2.svg.png)
> 
> 
> I'm struggling to locate official information from mechkeys.ca regarding this PCB, so images are best that's available 🫤

MechKeys.ca shut down a couple years ago, and as near as I know its proprietor completely left the hobby.

If I overlay your ABNT2 reference image over the PCB photo, it looks like the layout might be supported *if* the user uses the 1u/1u split Left Shift, but not with the normal 1.25u/1u split.

I'd also found a [post on reddit](https://www.reddit.com/r/MechanicalKeyboards/comments/6u6y3n/help_flashing_qmk_on_acr60_linux/) where a couple users implied/stated that flashing DZ60 firmware to an ACR60 works, so I compared the existing matrix pins and layouts (the ones that were present before my additions) and everything matched up. I know that MechKeys PCBs were frequently based on DZ designs - the PCB designer used to be active around QMK and I was able to [confirm that directly with them some years ago](https://github.com/qmk/qmk_firmware/pull/7583#pullrequestreview-329606134), so the ACR60 being a modified DZ60 makes sense to me.

I'll set this to WIP and work out some more accurate details.

**Comment by dunk2k** [comment]
On the, admittingly unrelated to this PR, mention of the DZ60 and `60_abnt2` Community Layout:
I've removed `60_abnt2` layout for keyboards `dz60` and `dztech/dz60v2` in PRs 21832 and 22841 respectively, as named boards don't support the 1u-1.75u split Right Shift of ABNT2/JIS 🙈.  

A great visual example of a PCB with both 2u Right Shift and 1u-1.75u Right Shift is the [Play Keyboard 60 v3](https://www.play-keyboard.store/products/play-keyboard-60-v3-pcb) 😊

**Comment by noroadsleft** [comment]
I actually just came back to ask what you thought about leaving the layout in, but changing the key sizes to match what's actually supported on the PCB. It's the same number of keycodes and the same sequence.

**Comment by dunk2k** [comment]
> I actually just came back to ask what you thought about leaving the layout in, but changing the key sizes to match what's actually supported on the PCB. It's the same number of keycodes and the same sequence.

I appreciate you asking my thoughts on it 🙂
The Community Layout "purist" in me is saying they ain't fussed so long as `60_abnt2` is removed from `community_layouts` 😆
Me personally, yeah keep the layout; resize keys to match what's supported and rename layout (`LAYOUT_abnt2` perhaps?), no harm in that 🙂

**Review by dunk2k — APPROVED** [review]

File: keyboards/mechkeys/acr60/matrix_diagram.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+│30     │ 2u LShift                       2u RShift │3D     │
+└───────┘                                           └───────┘
+┌────┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬──────┬───┐
+│30  │31 │32 │33 │34 │35 │36 │37 │38 │39 │3A │3B │3D    │3E │ Standard with Split Shifts
+└────┴───┴───┴───┴───┴───┴───┴───┴───┴───┴───┴───┴──────┴───┘   1.25u/1u LShift + 1u/1.75u/1u RShift
+┌────────┐                                   ┌───┬──────────┐
+│30      │ 2.25u LShift                      │3B │3D        │ 2.75u RShift
+└────────┘                                   └───┴──────────┘
+                                             ┌──────┬───┬───┐
+                                             │3C    │3D │3E │ 1.25u/1u LShift + 1.75u/1u/1u RShift
```
Comment (dunk2k) [inline keyboards/mechkeys/acr60/matrix_diagram.md:28]:
```suggestion
                                             │3C    │3D │3E │ 1.75u/1u/1u RShift
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/mechkeys/acr60/matrix_diagram.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+└────┴────┴────┴────────┴────┴──────────┴───┴───┴───┴───┴───┘
+                Split Spacebar
+                [2.25/1.25/2.75] or [2.75/1.25/2.25]
+               ┌──────────┬────┬────────┐
+               │44        │46  │48      │
+               └──────────┴────┴────────┘
+                6.25u Spacebar
+               ┌────────────────────────┐
+               │46                      │
+               └────────────────────────┘
```
Comment (dunk2k) [inline keyboards/mechkeys/acr60/matrix_diagram.md:42]: repeats "Standard" bottom row layout
```suggestion
```

File: keyboards/mechkeys/acr60/matrix_diagram.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+│40  │41  │43  │46                      │4A  │4B  │4D  │4E  │ ├─ Standard
+└────┴────┴────┴────────────────────────┴────┴────┴────┴────┘ │    or
+┌─────┬───┬─────┬───────────────────────┬─────┬───┬───┬─────┐ │  Infinity/True HHKB
+│40   │41 │43   │46                     │4A   │4B │4D │4E   │ │    (same matrix)
+└─────┴───┴─────┴───────────────────────┴─────┴───┴───┴─────┘ ┘
+┌─────┬───┬────┬────────────────────────┬───┬───┬───┬───┬───┐
+│40   │41 │43  │46                      │4A │4B │4C │4D │4E │
+└─────┴───┴────┴────────────────────────┴───┴───┴───┴───┴───┘
+┌─────┬─────┬───────────────────────────┬───┬───┬───┬───┬───┐
+│40   │41   │46                         │4A │4B │4C │4D │4E │
```
Comment (dunk2k) [inline keyboards/mechkeys/acr60/matrix_diagram.md:53]:
```suggestion
│40   │41   │46                         │4A │4B │4C │4D │4E │ LWKL
```

File: keyboards/mechkeys/acr60/matrix_diagram.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+└─────┴─────┴───────────────────────────┴───┴───┴───┴───┴───┘
+┌─────┬───┬─────┬───────────────────────────┬─────┬───┬─────┐
+│40   │41 │43   │46                         │4B   │4D │4E   │ Tsangan/WKL/HHKB
+└─────┴───┴─────┴───────────────────────────┴─────┴───┴─────┘
```
Comment (dunk2k) [inline keyboards/mechkeys/acr60/matrix_diagram.md:57]:
```suggestion
└─────┴───┴─────┴───────────────────────────┴─────┴───┴─────┘
┌─────┬───┬─────┬───────────────────────────┬───┬───┬───┬───┐
│40   │41 │43   │46                         │4B │4C │4D │4E │ Tsangan Arrow
└─────┴───┴─────┴───────────────────────────┴───┴───┴───┴───┘
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/mechkeys/acr60/keyboard.json
Comment (dunk2k) [inline keyboards/mechkeys/acr60/keyboard.json:1]: layout `LAYOUT_60_true_hhkb` is missing 🙈

**Review by dunk2k — CHANGES_REQUESTED** [review]

**Review by dunk2k — APPROVED** [review]

**Pull request merged**